.travis_script.sh

```diff
@@ -70,7 +70,20 @@ then
 fi
 export PYTHONPATH=$(pwd)/tools/pylib/:$PYTHONPATH
 
-time make $MAIN_TARGET|| exit
+time make $MAIN_TARGET
+make_exit=$?
+if [[ $make_exit -gt 0 ]]; then
+    make clean > /dev/null
+    echo -e $RED_FG
+    echo "**************************************************"
+    echo "Printing make commands:"
+    echo "**************************************************"
+    echo -e $RESET_FG
+    echo
+    make -n $MAIN_TARGET
+    exit $make_exit
+fi
+
```

Comment:
I added this to try and debug this particular branch, but it may be more generally useful. Happy to remove it if there are any objections